### PR TITLE
feat(RHTAPREL-298): remove ReleaseStrategy

### PR DESCRIPTION
**Pull request by davidmogar**

### CODE REVIEW

**Comment by davidmogar** [comment]
> @davidmogar can you please add the jira reference in PR title (I see it's included in the first commit's title). And also the description. Also, the second commit doesn't conform to gitlint rules.

Taht's just for review purposes. I'll fix the commits before merging.

File: api/v1alpha1/releaseplan_types.go
Comment (mmalina) [inline api/v1alpha1/releaseplan_types.go:37]: This is the first time I hear of this. Where is this defined? I don't see this in the linked Jira. Also, maybe the PR description / commit message could say a bit more about these changes as well?

Comment (davidmogar) [reply api/v1alpha1/releaseplan_types.go:37]: This is what you wrote but just moved to the shared-utils repo.

File: .idea/misc.xml
Comment (davidmogar) [reply .idea/misc.xml:1]: Yeah, in some repos the whole directory was ignored but not in this one. I'll do it in a different PR as it doesn't affect the IDE at all.

File: api/v1alpha1/releaseplan_types.go
Comment (davidmogar) [reply api/v1alpha1/releaseplan_types.go:37]: Oh, right. I didn't get your comment. This is a feature that is coming. The code for it will be added with a different ticket but I wanted to do all the API changes on the same PR.

Comment (mmalina) [reply api/v1alpha1/releaseplan_types.go:37]: ok. but this PR still has no description. could we have at least a brief summary of what is being done here besides just removing ReleaseStrategy?

File: api/v1alpha1/releaseplanadmission_types.go
Comment (johnbieren) [inline api/v1alpha1/releaseplanadmission_types.go:63]: Is it possible to print a list here? Having the application names show here would be quite nice

**Review by johnbieren — COMMENTED** [review]
Biggest thing for me was that I wasn't expecting this PR to be moving to shared-utils. I thought we were revisiting if we were going to use that repo. Not 100% sure I didn't miss anything, but if I did, surely the e2e tests will catch it

File: api/v1alpha1/releaseplan_types.go
Comment (davidmogar) [reply api/v1alpha1/releaseplan_types.go]: As I commented with you, the final version of the whole change was already implemented (the one including shared-utils) so I had to revert all of that. This I just simply forget about it. I restored the package and left the named imports so it's easy to go back to shared-utils later on.

File: api/v1alpha1/releaseplan_types.go
Comment (davidmogar) [reply api/v1alpha1/releaseplan_types.go:37]: I think that it's fine here as we talk about a pipeline to be executed before the release pipeline... Later on, when it's finally implemented, we will have to add a name for sure... In my code I used `dev` and `managed` instead of `release`. But that code is reverted and will come in a separate PR.

File: api/v1alpha1/releaseplanadmission_types.go
Comment (davidmogar) [reply api/v1alpha1/releaseplanadmission_types.go:63]: No. This is the info printed by kubectl in a table. We would break the table easily. It will have to be a detailed view.

File: api/v1alpha1/releaseplan_types.go
Comment (davidmogar) [reply api/v1alpha1/releaseplan_types.go:37]: As soon as I fix the commits after an initial review.

**Comment by johnbieren** [comment]
Why did you add this `Use fork` commit?

File: api/v1alpha1/releaseplan_types.go
Comment (johnbieren) [reply api/v1alpha1/releaseplan_types.go]: Then if, after review, we decide not to use shared-utils (as is my preference) we have all these imports changed to `shared` and it not making much sense. Why not do keep all the shared-utils stuff in one PR, instead of half now half later

**Comment by davidmogar** [comment]
> Why did you add this `Use fork` commit?

We did it in the previous PR as well to point to the fork. It not I couldn't get the code in E2E. The whole commit will be deleted before merging this.

**Comment by johnbieren** [comment]
> > Why did you add this `Use fork` commit?
> 
> We did it in the previous PR as well to point to the fork. It not I couldn't get the code in E2E. The whole commit will be deleted before merging this.

I've never had to make a commit like that for the pairing. Maybe I am missing something?

**Comment by davidmogar** [comment]
> > > Why did you add this `Use fork` commit?
> > 
> > 
> > We did it in the previous PR as well to point to the fork. It not I couldn't get the code in E2E. The whole commit will be deleted before merging this.
> 
> I've never had to make a commit like that for the pairing. Maybe I am missing something?

Maybe I'm doing the imports in the wrong way, but it could be also because the new PipelineRef stuff. It also affects API changes. If you don't change the API not much of a problem. In any case, not a big deal. It will be removed any way.

**Review by johnbieren — COMMENTED** [review]
The PR LGTM once the `s/redhat-appstudio/davidmogar/g` switch is removed. When you remove that, can you also remove the `.idea/misc.xml` file from the PR too please? Then I will be good to approve

**Review by johnbieren — APPROVED** [review]

**Comment by ralphbean** [comment]
:partying_face:

**Pull request merged**
